js/src/mapCollector.js

```diff
@@ -54,39 +54,72 @@ export class MapCollector {
         })
         break
       case BreakType.DOUBLE:
-        this.map.push({
-          type: MapType.BREAK,
-          double: true,
-        })
+        let paragraphBreakAdded = false
+        // iterate through map backwards:
+        for (let i = this.map.length - 1; i >= 0; --i) {
+          const map = this.map[i]
+          if (map.type === MapType.BREAK && map.double) {
+            paragraphBreakAdded = true
+            break
+          } else if (!this.isSingleBreak(map)) {
+            break
+          }
+        }
+        if (!paragraphBreakAdded) {
+          this.map.push({
+            type: MapType.BREAK,
+            double: true,
+          })
+        }
         break
     }
 
     this.lastBreak = BreakType.NONE
   }
 
-  processText() {
+  isSingleBreak(mapObject) {
+    const isSingleBreak = mapObject.type === MapType.BREAK && !mapObject.double
+    const isNewLine = mapObject.type === MapType.TEXT && mapObject.content === '\n'
+    return isSingleBreak || isNewLine
+  }
+
+  processText(trimEndSpaces = true) {
     if (this.text.length === 0) {
       return
     }
 
     const joinedText = this.text.map((element) => element.string).join('')
-    // TODO: might have to check for null string here
-    const trimmed = trimBeginAndEnd(joinedText)
-
-    if (!trimmed) {
-      // Trimmed into an empty string
-      //  Preserve all preceding breaks
-      this.text = []
-      return
+    let fullText = joinedText
+    if (trimEndSpaces) {
+      const trimmed = trimBeginAndEnd(joinedText)
```

Comment:
wouldn't it break the original logic since it'd been the default behaviour and now this code runs only if `trimEndSpaces` flag is set to true?
If this is the case the major version of the package should be bumped

Reply:
the default value of trimEndSpaces is true so this function call doesn't change unless false is specifically passed in.

Also since this function is not accessible by any package using the degausser we don't need to do a major version change anyways.